common/src/main/java/com/microsoft/identity/common/components/AndroidEncryptedStorageSupplier.kt

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.components
+
+import android.content.Context
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager
+import com.microsoft.identity.common.internal.util.SharedPrefStringNameValueStorage
+import com.microsoft.identity.common.internal.util.SharedPreferenceLongStorage
+import com.microsoft.identity.common.java.cache.IMultiTypeNameValueStorage
+import com.microsoft.identity.common.java.crypto.StorageEncryptionManager
+import com.microsoft.identity.common.java.interfaces.INameValueStorage
+import com.microsoft.identity.common.java.interfaces.IStorageSupplier
+
+class AndroidEncryptedStorageSupplier(private val context: Context,
+                                      private val storageEncryptionManager: StorageEncryptionManager)
+    : IStorageSupplier {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> getNameValueStore(storeName: String, clazz: Class<T>): INameValueStorage<T> {
+        val mgr: IMultiTypeNameValueStorage =
+            SharedPreferencesFileManager.getSharedPreferences(context, storeName, storageEncryptionManager)
+        if (Long::class.java.isAssignableFrom(clazz)) {
+            return (SharedPreferenceLongStorage(mgr) as INameValueStorage<T>)
```

Comment:
Are we using any if these?
`SharedPreferenceLongStorage` or `SharedPrefStringNameValueStorage`
I think we only have the String values, if that is the case, we should remove the other one, to keep it simple

Reply:
we use SharedPreferenceLongStorage, but we're not getting it from here...